### ISSUE TITLE
Hide messages displaying changes to room avatar/name/topic and other loud messages …

### ISSUE THREAD

**Issue by fooness** [vector-im/riot-web#13427]
It would be great to be able to optionally hide all messages like changes to room avatar/name/topic, adding widgets, et cetera … bascically everything that’s distracting and what’s not a message from users.

**Comment by ingingin** [comment]
Yes please! Or even better, for privacy, is to remove it completely. Or add a toggle in settings to allow your account to display these messages for everyone in a room.

Being a perfectionist like myself can I sit there all day spamming rooms with these messages by changing profile pictures and names til I get it completely right

I think it is important to keep the 
**`[username] joined the room.`**
and
**`[username] left the room.`**

**Comment by t3chguy** [comment]
![image](https://user-images.githubusercontent.com/2403652/82305517-a2f02980-99b5-11ea-8827-980c4fa4288d.png)

**Comment by t3chguy** [comment]
> Or even better, for privacy, is to remove it completely.

Its not possible to remove the events which cause room actions *for privacy*
They cause effect so they must happen.
Whether one client shows them by default is not "privacy"

If you change your name, the only way other users' clients get that update is via that event, whether or not it is shown in the timeline, it is not a thing at all conflicting with privacy.

**Comment by Bubu** [comment]
Got this as new user feedback yesterday: "What? It told everyone in every room that I changed my avatar?!?! That's terrifying!"

I understand that the events must end up at all other clients, but I disagree with the conclusion that this is unrelated to privacy.
Default behaviour is important here: There's a huge difference between this being a default hidden event where realistically no-one will look at your new avatar (until maybe when you speak next) and the avatar change being announces in 20 rooms and everyone browsing the timeline and being bored enough will click on the image.

Related but different: I definitely refrained from changing my name/avatar in the past because I didn't want to spam all the rooms I'm in with a new visible event.

**Comment by t3chguy** [comment]
> There's a huge difference between this being a default hidden event where realistically no-one will look at your new avatar (until maybe when you speak next) and the avatar change being announces in 20 rooms and everyone browsing the timeline and being bored enough will click on the image.

I disagree, there is many different angles to it including accessibility and a different privacy angle.

People may associate avatars to the user they are speaking to, if that avatar changes they ought to know so they know to learn the new association, this is especially important for accessibility purposes but even if you consider it from the privacy angle where you associate X has Y avatar and you might not realise that you are still speaking to X without explicitly being told their avatar changed.

**Comment by MadLittleMods** [comment]
Related to https://github.com/vector-im/element-web/issues/1544
